python/pyarrow/tests/strategies.py

```diff
@@ -35,7 +38,10 @@
         import tzdata  # noqa:F401
     except ImportError:
         zoneinfo = None
-import numpy as np
+try:
+    import numpy as np
+except ImportError:
+    np = None
```

Comment:
The "strategies" defined in this file are not marked as needing numpy. Are the tests using those strategies marked as such? 

(not that we are testing this, because we run the hypothesis tests only in one build which has numpy, so it's also fine to assume that numpy is available for the hypothesis tests I think)

Reply:
I've run hypothesis tests locally and have marked the ones that require numpy explicitly